Mathlib/Order/Monotone/Basic.lean

```diff
@@ -952,6 +952,11 @@ function `ℕ → α`. -/
 theorem exists_strictAnti [Nonempty α] [NoMinOrder α] : ∃ f : ℕ → α, StrictAnti f :=
   exists_strictMono αᵒᵈ
 
+lemma pow_self_mono : Monotone fun n : ℕ ↦ n ^ n := by
```

Comment:
Do you think we should have the `StrictMonoOn` (or perhaps `PNat`) version too?

Another possibly interesting statement is
```lean
theorem pow_monotoneOn : MonotoneOn (fun p : ℕ × ℕ ↦ p.1 ^ p.2) {p | p.1 ≠ 0} := by
```

bors d+

Reply:
I've added `pow_self_strictMonoOn` and `pow_monotoneOn`